snet_cli/utils.py

```diff
@@ -83,3 +85,40 @@ def type_converter(t):
         return web3.Web3.toChecksumAddress
     else:
         return str
+
+
+def _validate_path(path, entry_path):
+    return entry_path.parent in path.parents
+
+
+def _add_next_paths(path, entry_path_parents, walked_paths, next_paths):
+    with open(path) as f:
+        for line in f:
+            if line.strip().startswith("import"):
+                import_statement = "".join(line.split('"')[1::2])
+                if not import_statement.startswith("google/protobuf"):
+                    import_statement_path = PurePath(Path(path.parent.joinpath(import_statement)).resolve())
+                    if _validate_path(import_statement_path, entry_path_parents):
+                        if import_statement_path not in walked_paths and import_statement_path not in next_paths:
+                            next_paths.append(import_statement_path)
+                    else:
+                        raise ValueError("Path must not be a parent of entry path")
+
+
+def walk_imports(entry_path):
+    walked_paths = set()
+    next_paths = [entry_path]
+    while next_paths:
+        path = next_paths.pop()
+        if os.path.isfile(path):
+            walked_paths.add(path)
```

Comment:
The issue is "fixed" here, but I think the naming of the variables is tripping you up. I ask the following question:

Why can't you add every file to what is called `walked_paths` at the same time you add them to what is called `next_paths`? E.g. initialize walked_paths to `set([entry_path])` at line 109, and `walked_paths.add(import_statement_path)` at line 104? Then line 102 can just check the `walked_paths` set rather than both.

Reply:
Fixed